api-2.0/src/test/java/org/openmrs/module/fhir/api/RelatedPersonServiceTest.java

```diff
@@ -120,4 +139,30 @@ public void updateRelatedPerson_shouldCreateIfNotExists() throws Exception {
 		assertEquals(relationship.getPersonA(), updatedRelationship.getPersonA());
 		assertEquals(relationship.getPersonB(), updatedRelationship.getPersonB());
 	}
+
+	@Test
+	public void searchRelatedPersonByIdentifier_shouldReturnResourceIfExists() {
+		List<RelatedPerson> relatedPersonList = getService().searchRelatedPersonByIdentifier(PATIENT_IDENTIFIER);
+
+		assertEquals(relatedPersonList.size(), 3);
+		assertNotNull(relatedPersonList);
+		assertEquals(relatedPersonList.get(0).getGender().getDisplay(), "Male");
+		assertEquals(relatedPersonList.get(0).getId(), "b70cb368-cf80-4a22-9d75-5d0d1f9b75dc");
+		assertEquals(relatedPersonList.get(0).getPatient().getReference(), "person/ba1b19c2-3ed6-4f63-b8c0-f762dc8d7562");
+
+		assertEquals(relatedPersonList.get(1).getGender().getDisplay(), "Male");
+		assertEquals(relatedPersonList.get(1).getId(), "c18717dd-5d78-4a0e-84fc-ee62c5f0676a");
+		assertEquals(relatedPersonList.get(0).getPatient().getReference(), "person/ba1b19c2-3ed6-4f63-b8c0-f762dc8d7562");
```

Comment:
Is there any reason why you are not putting this assertion of line 155 close to the ones of 149 to 151?

Reply:
Did you also just forget to assert getReference for the values at index 1 and 2?